### PR TITLE
MAT-7790: add function tab to cql builder

### PR DESCRIPTION
**Pull request by chubert-sb**
## MADiE PR

Jira Ticket: [MAT-7790](https://jira.cms.gov/browse/MAT-7790)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/CqlBuilderPanel/functionsSection/functionBuilder/FunctionBuilder.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  TextArea,
+  TextField,
+} from "@madie/madie-design-system/dist/react";
+import "../Functions.scss";
+import ExpandingSection from "../../../common/ExpandingSection";
+import { Checkbox, FormControlLabel } from "@mui/material";
+import { Box } from "@mui/system";
+import ConfirmationDialog from "../../common/ConfirmationDialog";
+
+export interface Funct {
```
Comment (adongare) [inline src/CqlBuilderPanel/functionsSection/functionBuilder/FunctionBuilder.tsx:16]: is there a name clash? if not can we name it `Function`?

Comment (chubert-sb) [reply src/CqlBuilderPanel/functionsSection/functionBuilder/FunctionBuilder.tsx:16]: yeah it seemed like `function` was a reserved name. I couldn't think of anything better than `funct`. I can change it if anyone can think of something better

**Review by adongare — APPROVED** [review]
